### PR TITLE
Updates unicorn to work with Sinatra Apps, and provide ENV variables to the application

### PR DESCRIPTION
**Pull request by crowdmatt**

### CODE REVIEW

**Comment by rnhurt** [comment]
+1

This looks like it would solve my problem.  I'm trying to run a Sinatra app on a Rails layer and everything works except for the database connection.  Something in there is not configured correctly and Unicorn is throwing an error:

```
E, [2013-05-14T18:44:47.519395 #17538] ERROR -- : ActiveRecord::AdapterNotSpecified (ActiveRecord::AdapterNotSpecified)
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/activerecord-3.2.13/lib/active_record/connection_adapters/abstract/connection_specification.rb:24:in `spec'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/activerecord-3.2.13/lib/active_record/connection_adapters/abstract/connection_specification.rb:130:in `establish_connection'
/srv/www/thumbnail_web/shared/config/unicorn.conf:49:in `block in reload'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/unicorn-4.6.2/lib/unicorn/http_server.rb:591:in `call'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/unicorn-4.6.2/lib/unicorn/http_server.rb:591:in `init_worker_process'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/unicorn-4.6.2/lib/unicorn/http_server.rb:615:in `worker_loop'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/unicorn-4.6.2/lib/unicorn/http_server.rb:500:in `spawn_missing_workers'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/unicorn-4.6.2/lib/unicorn/http_server.rb:511:in `maintain_worker_count'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/unicorn-4.6.2/lib/unicorn/http_server.rb:277:in `join'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/gems/unicorn-4.6.2/bin/unicorn_rails:209:in `<top (required)>'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/bin/unicorn_rails:23:in `load'
/home/deploy/.bundler/thumbnail_web/ruby/1.9.1/bin/unicorn_rails:23:in `<main>'
```

File: unicorn/attributes/default.rb [hunk 1]
```diff
@@ -1,6 +1,18 @@
-include_attribute 'rails::rails'
+if node[:sinatra] || node[:web_application_type] == 'sinatra'
+  include_attribute 'sinatra::default'
+elsif node[:rails] || node[:web_application_type] == 'rails'
+  include_attribute 'rails'
```
Comment (ujuettner) [inline unicorn/attributes/default.rb:4]: Should be: include_attributes 'rails::rails'

File: unicorn/attributes/default.rb [hunk 2]
```diff
@@ -1,6 +1,18 @@
-include_attribute 'rails::rails'
+if node[:sinatra] || node[:web_application_type] == 'sinatra'
+  include_attribute 'sinatra::default'
```
Comment (ujuettner) [inline unicorn/attributes/default.rb:2]: AFAIK, will throw an exception, as there is no cookbook named sinatra.

**Comment by ujuettner** [comment]
Thanks a lot for your contribution. I've just tried to merge your changes and stumbled upon some minor issues - commented on the corresponding lines. BTW, setup works with the recipes assignments mentioned in the README, but how do you want to trigger a deployment? The OpsWorks deploy recipes aren't triggered, if the application_type is set to 'sinatra'. Do you expect this being handled in a custom cookbook?

**Comment by moorage** [comment]
Hey -- yes, this custom cookbook is in: https://github.com/crowdmob/chef-sinatra

However, I've come to realize that very few people will be using sinatra come Rails 4.  Happy to have this closed without merging =[

**Comment by ujuettner** [comment]
Ok, that was the missing link for me. Actually I would merge your changes on unicorn/templates/default/unicorn.conf.erb and a slightly different change on your changes on unicorn/templates/default/unicorn.service.erb, which recognizes the web_application_type (this is then handed over in unicorn/recipes/rails.rb). By that, a Rails/Rack app can be used the use of the matching unicorn script is switched by web_application type - the deployment code is easily re-used and the configuration is simpler.
So, by setting {"web_application_type": "sinatra"} within the custom JSON unicorn is used, otherwise unicorn_rails.
Are you feeling ok with that?

**Comment by moorage** [comment]
Got it -- I probably can't prioritize this since we're working on Go and
not sinatra anymore :)

Best,

## Matt

http://www.linkedin.com/in/matthewpaulmoore
(650) 888-5962

**Comment by githuesch** [comment]
This sounds to me like this request can be closed without being merged. @moorage What do you think?

**Comment by moorage** [comment]
@githuesch sure, that's fine!

**Pull request closed without merging**